File: mad3004-introd-swift/C0711561_F2017_MAD3004_FinalProject/C0711561_F2017_MAD3004_FinalProject/Vehicle.swift
public class Vehicle : IPrintable {

    private var make: String
    private var model: String
    private var plate: String
    
    init() {
        make = ""
        model = ""
        plate = ""
    }
    
    init(pMake: String, pModel: String, pPlate: String) {
        make = pMake
        model = pModel
        plate = pPlate
    }
    
    func getVehicleType() -> String {
        fatalError("This method must be overridden")
    }
    
    func printMyData() -> String {
        return "  -Make: \(make)\n  -Model: \(model)\n  -Plate: \(plate)"
    }

}

File: mad3464-programming-java-i/C0711561_F2017_MAD3464_FinalProject/src/c0711561/f2017/mad3464/finalproject/vehicle/Vehicle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package c0711561.f2017.mad3464.finalproject.vehicle;

import c0711561.f2017.mad3464.finalproject.IPrintable;

/**
 *
 * @author macstudent
 */
public abstract class Vehicle implements IPrintable {
    
    private String make;
    private String model;
    private String plate;

    public Vehicle(String make, String model, String plate) {
        this.make = make;
        this.model = model;
        this.plate = plate;
    }

    public String getMake() {
        return make;
    }

    public String getModel() {
        return model;
    }

    public String getPlate() {
        return plate;
    }
    
    public abstract String getVehicleType();

    @Override
    public String printMyData() {
        return String.format("   -Make: %s\n  -Model: %s\n -Plate: %s", make, model, plate);
    }
}
